.github/workflows/create-release-main.yml

```diff
@@ -0,0 +1,71 @@
+name: Create new preview release (main)
+
+on:
+  # run when pushed to main is published,
+  # which creates a new tag
+  push:
+    branches:
+      - main
```

Comment:
IMO this should be every 24 hours or every weekend.

Reply:
Every 24h or every week is another option, but it feels like main images are more ideal given you don't want to wait to try something exciting that was contributed. But that is just my opinion of course, I am not a maintainer :) If we do not go with main, which is fairly common, then we should go with daily and not weekly as that is too long.

Additionally, I don't see the benefit of pushing a new version if there are no changes.

Manual trigger is something I'm planning on contributing once this gets merged, but from a non-main branch feels only reasonable scenario

Reply:
Manual trigger is useful for testing purposes, esp when someone is working on changing/fixing this Github Action.

Triggering on pushes to `main` is probably the best approach - I agree that weekly is too long, and daily has a bunch of downsides (two being new images even when there are no changes, enforced waiting to try things out).

However, lots of things push to `main` that wouldn't impact on the image, so I think we need some path filtering as well. E.g. we don't need a new image if our documentation is updated under `/docs/`, nor if there's a bug fix or security update for ASO v1.

Reply:
> Manual trigger is useful for testing purposes, esp when someone is working on changing/fixing this Github Action.

As mentioned above, I was planning on doing that as a follow-up PR to make the change more explicit and easier to review given we may want to handle things differently, is it OK to do it in a new PR?

We (APIM team) have the same plans, was just not added yet.

> However, lots of things push to main that wouldn't impact on the image, so I think we need some path filtering as well. E.g. we don't need a new image if our documentation is updated under /docs/, nor if there's a bug fix or security update for ASO v1.

That definitely makes sense to me, but I was not 100% sure which filters to use. Would you mind sharing the ones you think make sense or shall I just exclude docs as the rest all feel related?

Reply:
Added filter to exclude docs for now.

Reply:
Tracking build from non-main branch in https://github.com/Azure/azure-service-operator/issues/3713

Reply:
> As mentioned above, I was planning on doing that as a follow-up PR to make the change more explicit

I think we may be talking about two different things. I was thinking about adding an additional trigger to your workflow:

``` yaml
on:
  workflow_dispatch:
  # no content, allows manual triggering
```

This would allow repo contributors to directly trigger the workflow as needed.

Reply:
> Added filter to exclude docs for now.

I'd invert it to only rebuild the image if something under the `v2` folder is modified.

Reply:
Valid point, I thought you wanted to have customer tags; added both!